### ISSUE TITLE
Function calc_connect() is still called with cutoff of zero

### ISSUE THREAD

**Issue by max-veit** [libAtoms/QUIP#11]
There are many cases where one might want to use a potential with a zero cutoff. For instance, in `FilePot.f95`, the [function cutoff()](https://github.com/libAtoms/QUIP/blob/public/src/Potentials/FilePot.f95#L209) always returns zero. This causes problems whenever [`quippy.Potential.update()`](https://github.com/libAtoms/QUIP/blob/public/quippy/quippy/potential.py#L322) is called, since that function sets the attached `Atoms` object's cutoff to the value of `self.cutoff()`, which is zero. Two lines later it calls `atoms.calc_connect()`, which throws an error due to the [check in `Connection::connection_calc_connect()`](https://github.com/libAtoms/QUIP/blob/public/src/libAtoms/Connection.f95#L1032) that the cutoff not be zero.

I only recently noticed this because of the combination of using a FilePot and calling a function (`atoms.set_calculator(pot)`) that triggers `quippy.potential.update()`. There are a few fixes I can think of:
- Don't throw an error if `Connection::calc_connect()` is called with a zero cutoff. This would make me happy because I've come into numerous other situations where, for one reason or another, the cutoff is set to zero but `calc_connect()` is still erroneously called. However, I'm assuming that error was there for a reason, so there may be other unforeseen consequences of demoting this error to a warning.
- Don't call `calc_connect()` in `quippy.potential.update()` if the cutoff is zero. This is a simple fix, but there may be other places in quippy where `calc_connect()` is erroneously called on a zero cutoff.
- Don't change the cutoff in `quippy.potential.update()` if the the attached `Atoms`'s cutoff is greater than the `Potential`'s cutoff (indicating that the `Atoms` cutoff was manually set earlier).

Any input would be appreciated.

**Comment by jameskermode** [comment]
Throwing an error from calc_connect() when it's called with a zero cutoff is useful to help identify mistakes where the cutoff has not been set (since the default value of atoms%cutoff is zero). With the current defaults, changing quippy.Potential to that it doesn't call calc_connect() if the cutoff is zero would be the correct fix.

An alternative would be to change the default value of atoms%cutoff from 0 to -1 and throw an error in calc_connect() if cutoff < 0, but return immediately without error if cutoff == 0. This would allow zero cutoffs to be set explicitly without losing the nice feature of raising an error  when no cutoff has been assigned.

Either solution would be OK.

**Comment by gabor1** [comment]
actually James I like the second solution, the -1 would catch the unset cutoffs, but explicitly set zero cutoffs would be OK, and it would save a lot of trouble. of course we totally carpet-bombed the code by now with if(cutoff > 0) call calc_connect() lines⋅⋅

**Comment by jameskermode** [comment]
Great - Max, would you be able to try this out and fix the resulting failing tests?

**Comment by max-veit** [comment]
I haven't tracked down where the default is actually set - I thought it would be in [`Atoms_types.f95`](https://github.com/libAtoms/QUIP/blob/9080115cdb00425b4f46d27cb2df01a32839bf69/src/libAtoms/Atoms_types.f95#L247), but changing that doesn't change the cutoff of a newly created Atoms object in Python (I've tried an empty constructor as well as reading from an XYZ file). I've also poked around in some of the `quippy` and `ase` Atoms constructors - am I missing something?

Edit: Indeed I was missing something - I didn't have my recompiled `quippy` properly installed. I'll rerun the tests and report if anything fails.
